registry/interchange.go

```diff
@@ -45,12 +50,22 @@ func MakeJobInterchange(j amboy.Job, f amboy.Format) (*JobInterchange, error) {
 		return nil, err
 	}
 
+	status := j.Status()
+	truncatedErrs, isTruncated := truncateJobErrors(status.Errors)
+	status.Errors = truncatedErrs
+	grip.WarningWhen(isTruncated, message.Fields{
```

Comment:
do we log status.Errors elsewhere then? I'm skeptical that we should be logging truncated_errs twice. Does this error seem helpful in general? Also should we log this at a debug level if we want to keep it?

Reply:
We need this as a monitoring mechanism because if we don't log the truncated errors here, we'll never be able to tell that there was a problem. Because the error message is so long, the generic job log [here](https://github.com/mongodb/amboy/blob/f84bf0b328aee41ff97a31d9a7380a2b8f079a51/pool/helpers.go#L92-L103) will never log to Splunk (for the reason stated in the description), so we'll have no way of telling that the job really completed, nor what errors there were.

Reply:
Should the truncating happen at the time of executing the job instead then? That way the logs will be consistent regardless of the size of error, and then when we get to the job interchanging the status will already be a reasonable size (I think)

Reply:
I mean, I guess we could truncate while doing `AddErrors`, but that would also result in weirdness because we wouldn't have a way of informing the user (unless we logged in `AddErrors`, which seems even noisier than just logging when the job is saved back to the DB).

In any case, this is an edge case that only gets hit specifically by generate.tasks in a case where the YAML has so many errors that it emits 16 MB worth of them. It's rare enough that we've only seen this bug occur once in 5 years. I don't think that it's worth putting a lot of effort into handling an edge case for a specific job.